src/NServiceBus.Core/Pipeline/satellitePipelineExecutor.cs

```diff
@@ -17,7 +14,12 @@ public SatellitePipelineExecutor(IBuilder builder, SatelliteDefinition definitio
 
         public Task Invoke(MessageContext messageContext)
         {
+            messageContext.Context.Set(messageContext.TransportTransaction);
```

Comment:
Correct. If a component has direct access to `MessageContext.TransportTransaction`, it should use it. Not all components do, though (e.g. persisters).